src/tools/custom_exceptions.h

```diff
@@ -0,0 +1,32 @@
+#ifndef INCLUDE_TOOLS_CUSTOM_EXCEPTION_H
+#define INCLUDE_TOOLS_CUSTOM_EXCEPTION_H
+
+#include <iostream>
+#include <exception>
+#include <string>
+
+class CustomException : public std::exception {
```

Comment:
You might want to call this something like CUException (Clusters Ultimate exception) or similar. "CustomException" is a bit too generic IMO.

Reply:
Agreed. I will change the name to CUException in the next push.